broker-core/src/test/java/io/zeebe/broker/job/ActivateJobsTest.java

```diff
@@ -95,9 +97,7 @@ public void shouldRejectInvalidAmount() {
             .sendAndAwait();
 
     // then
-    assertThat(response.getRejectionType()).isEqualTo(RejectionType.BAD_VALUE);
-    assertThat(response.getRejectionReason())
-        .isEqualTo("Job batch amount must be greater than zero, got 0");
+    assertThat(response.getRejectionType()).isEqualTo(RejectionType.INVALID_ARGUMENT);
```

Comment:
I think it is not bad to have the assertions for the error messages, because that way you can distinguish between different errors and also can check if you implemented correctly. If we change the impl we also get faster feedback. My thoughts on this.

Reply:
I feel matching strings is kind of brittle, unless we go hardcore on enforcing a single pattern and consider it a regression to change it. I figured matching the code is probably enough to distinguish the class of errors in most cases, perhaps we'd need a more granular codification? Then again if we have a single point of error formatting perhaps it's not too bad. Still, it feels like we'd be coupling tests that test feature A to the formatting of errors (i.e. changing the format means changing a whole lot of tests potentially).